### PR TITLE
Add GitHub action for running unit tests

### PR DESCRIPTION
**Pull request by lowiebenoot**
### Description

Added a GitHub action for running unit tests.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+name: 'Test'
+on: push
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
```
Comment (YorickH) [inline .github/workflows/test.yml:6]:
```suggestion
    runs-on: self-hosted
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install dependencies
+        run: yarn
```
Comment (YorickH) [inline .github/workflows/test.yml:10]: Add caching for `yarn`

File: .github/workflows/test.yml (hunk 1 above)
Comment (lowiebenoot) [reply .github/workflows/test.yml:6]: Why actually?

Comment (YorickH) [reply .github/workflows/test.yml:6]: You can leave it, these small ones are not a real issue

**Review by tiagobastos — APPROVED** [review]

**Review by YorickH — APPROVED** [review]

**Pull request merged**